packages/ui/test/common/components/ConnectionStatus.test.tsx

```diff
@@ -44,11 +44,10 @@ describe('UI: Connection status component', () => {
   })
 
   it('Disconnected', async () => {
+    useApi.connectionState = 'disconnected'
     renderComponent()
 
     act(() => {
-      eventEmitter.emit('connected')
-      useApi.connectionState = 'disconnected'
       eventEmitter.emit('disconnected')
     })
```

Comment:
@p-sad take a look at this part please. Is it okay?

Reply:
This one is weird, changing from connected to disconnected seems to work perfectly well in browser but doesn't seem to register in tests. It seems like it has a problem with animation/timeout, since the below version passes tests:
```ts
  it('Disconnected', async () => {
    renderComponent()

    act(() => {
      eventEmitter.emit('connected')
      jest.runOnlyPendingTimers()
    })

    act(() => {
      useApi.connectionState = 'disconnected'
      eventEmitter.emit('disconnected')
    })
  })
```

Reply:
Decided to commit this change, the fact that this works perfectly in manual tests and the test requires two separate `act` calls leads me to believe that this might be some sort of testing library quirk